src/ast/semantic_analyser.cpp

```diff
@@ -1779,11 +1783,20 @@ int SemanticAnalyser::create_maps(bool debug)
         Integer &min = static_cast<Integer&>(min_arg);
         Integer &max = static_cast<Integer&>(max_arg);
         Integer &step = static_cast<Integer&>(step_arg);
-        bpftrace_.maps_[map_name] = std::make_unique<bpftrace::Map>(map_name, type, key, min.n, max.n, step.n, bpftrace_.mapmax_);
+        bpftrace_.maps_[map_name] = std::make_unique<bpftrace::Map>(
+            map_name, type, key, min.n, max.n, step.n, bpftrace_.mapmax_);
+        bpftrace_.maps_[map_name]->id = bpftrace_.map_ids_.size();
+        bpftrace_.map_ids_.push_back(map_name);
```

Comment:
I'm not really happy with this. I think the long term fix should be to move the map storage out of bpftrace, for now this works ok